### PR TITLE
 DateInput with Actual Data

### PR DESCRIPTION
**Pull request by sarajaur**
# Related Issue

A Pull Request should close a **single** issue; multiple issues can be closed when the issues are **small and related**, but that should be an exception not the rule. Please keep Pull Requests small and targeted; large 'code drops' with dozens of files will be closed and asked to split into reviewable pieces. Reviews that need to be large due to dependencies will be
reviewed on a case-by-case basis.

Please use one of the well-known [github fixes keywords](https://help.github.com/en/articles/closing-issues-using-keywords) to reference
the issue fixed with this PR (eg Fixes #<github issue number>). If an issue doesn't yet exist please create one to aid
in issue tracking.

**NOTE**: For multiple issues resolved by this PR use the corresponding keywords **every time** in a comma-delimited list per the reference
page above.

# Description

For all Pull Requests, please describe how the issue was fixed or how the feature was implemented from a summary level. This information will be used to help provide context to the reviewers of the pull request and should be additive to the issue being closed.

- Implemented Date Input with actual data using the model based approach.
- https://app.vidcast.io/share/012c8663-4c3b-416c-93f2-74caf92151f3

# Sample Card

If appropriate, please include a link to a card in one of the samples directories that can be used to validate this change. This can be an existing card or a card added with this PR.

# How Verified

How you verified the fix, including one or all of the following:
1. New unit tests that were added if any. If none were added please add a quick line explaining why not.
2. Existing relevant unit/regression tests that you ran
3. Manual scenario verification if any; ***Do include .gif's or screenshots of the testing you performed here if you think that it
will aid in code reviews or corresponding fixes on other platforms for eg.***

### CODE REVIEW

File: source/qml_v2/AdaptiveCardQmlEngine/models/CollectionItemModel.h [hunk 1]
```diff
@@ -55,7 +56,7 @@ class CollectionItemModel : public QAbstractListModel
     void populateRowData(std::shared_ptr<AdaptiveCards::BaseCardElement> element);
     void populateTextBlockModel(std::shared_ptr<AdaptiveCards::TextBlock> textBlock, RowContent& rowContent);
     void populateImageModel(std::shared_ptr<AdaptiveCards::Image> image, RowContent& rowContent);
-    void populateRichTextBlockModel(std::shared_ptr<AdaptiveCards::RichTextBlock> richTextBlock, RowContent& rowContent);
+    void populateRichTextBlockModel(std::shared_ptr<AdaptiveCards::RichTextBlock> rightTextBlock, RowContent& rowContent);
     void populateDateInputModel(std::shared_ptr<AdaptiveCards::DateInput> dateInput, RowContent& rowContent);
```
Comment (soumykun) [inline source/qml_v2/AdaptiveCardQmlEngine/models/CollectionItemModel.h:60]: Why this change ? This is incorrect.

File: source/qml_v2/AdaptiveCardQmlEngine/models/DateInputModel.h [hunk 2]
```diff
@@ -2,6 +2,10 @@
 #include "AdaptiveCardContext.h"
 #include "DateInput.h"
 
+#include "OpenUrlAction.h"
+#include "ToggleVisibilityAction.h"
+#include "SubmitAction.h"
+
```
Comment (soumykun) [inline source/qml_v2/AdaptiveCardQmlEngine/models/DateInputModel.h:8]: Are these actions required ?

File: source/qml_v2/AdaptiveCardQmlEngine/models/DateInputModel.h [hunk 3]
```diff
@@ -11,8 +15,55 @@ class DateInputModel : public QObject
...
+    QString mDateInputFormat;
+    QString mDateFormat;
+    QString mInputMask;
+    QString mRegex;
+    QString mFormattedDateString;
+
+    bool mVisible;
+    bool mIsRequired;
+    bool mValidationRequired;
+
```
Comment (soumykun) [inline source/qml_v2/AdaptiveCardQmlEngine/models/DateInputModel.h:66]: Not initialized , the boolean variables ????

File: source/qml_v2/AdaptiveCardQmlEngine/models/DateInputModel.h (hunk 2 above)
Comment (sarajaur) [reply source/qml_v2/AdaptiveCardQmlEngine/models/DateInputModel.h]: No

**Review by soumykun — APPROVED** [review]

**Review by ramkrishnajana — APPROVED** [review]
